### PR TITLE
council - candidacy withdrawal fix

### LINKED ISSUE #2877: Council invalid Voting stage
The voting stage happens without more candidacies than the council size, if more candidacies than the council size are announce but some are then withdrawn.

### PR DESCRIPTION
**Pull request by ondratra**
Fixes #2877

### CODE REVIEW

File: runtime-modules/council/src/lib.rs [hunk 1]
```diff
@@ -1336,6 +1336,30 @@ impl<T: Trait> Mutations<T> {
...
+        stage_data: &CouncilStageAnnouncing,
+        membership_id: &T::MemberId,
+        candidate: &CandidateOf<T>,
+    ) {
+        // release candidacy stake
+        Self::release_candidacy_stake(&membership_id, &candidate.staking_account_id);
+
+        // prepare new stage
+        let new_stage_data = CouncilStageAnnouncing {
+            candidates_count: stage_data.candidates_count - 1,
```
Comment (shamil-gadelshin) [inline runtime-modules/council/src/lib.rs]: Please, use safe arithmetics (saturating_sub).

**Review by shamil-gadelshin — CHANGES_REQUESTED** [review]
Looks good.

Just a small comment about the safe arithmetics.

File: runtime-modules/council/src/lib.rs (hunk 1 above)
Comment (ondratra) [reply runtime-modules/council/src/lib.rs]: Ok, I've added safe arithmetics in a couple more places as well.

**Review by shamil-gadelshin — APPROVED** [review]
LGTM

**Review by thesan — APPROVED** [review]
Thanks

**Pull request merged**